weasyl/wsgi.py

```diff
@@ -54,7 +55,14 @@ def weasyl_404(request):
     wsgi_app = ProfilerMiddleware(
         wsgi_app, profile_dir=m.MACRO_STORAGE_ROOT + 'profile-stats')
 if config_obj.has_option('sentry', 'dsn'):
-    wsgi_app = mw.SentryEnvironmentMiddleware(wsgi_app, config_obj.get('sentry', 'dsn'))
+    sentry_sdk.init(
+        dsn=config_obj.get('sentry', 'dsn'),
+        release=d.CURRENT_SHA,
+        traces_sample_rate=float(config_obj.get('sentry', 'traces_sample_rate')),
+        integrations=[PyramidIntegration()],
+        send_default_pii=False,  # can’t be enabled as long as `before_send` doesn’t run for performance tracing!
```

Comment:
I looked over this earlier, and the comment is still breaking my brain (might just not be grokking something); the ``mw.strip_session_cookie`` function precludes this being ``True``, for performance reasons?

According to the [docs](https://develop.sentry.dev/sdk/data-handling/), one of the things this would permit is sending the user as Pyramid knows it; though we seem to override that by setting it ``set_user()`` on L99 in the middleware (this PR). It also seems to strip all cookies when ``False``, even though we sanitize ``WZL`` over in ``strip_session_cookie``.

Reply:
Not for performance reasons, for security reasons. `before_send` (i.e. `mw.strip_session_cookie`) runs for captured exceptions and messages, but *not* for performance tracing (enabled by the `traces_sample_rate` above). So with `send_default_pii=True`, performance tracing would have unredacted `WZL` cookies attached to it. The explicit `send_default_pii=False` is a workaround until there’s some way of filtering those events too.